### DISCUSSION TITLE
foreach() argument must be of type array|object, string given

### DISCUSSION THREAD

**Discussion by encoderuz** [spatie/laravel-permission#2354] [Q&A] [answered]
``Hello everyone! 
I'm new in laravel.
When I added this __constructor into my `PostController.php`
```
public function  __construct()
    { 
       $this->middleware(['role:admin','permission:create|edit']);
    }
```
occured this error => `foreach() argument must be of type array|object, string given`
```
Laravel 10 version
Laravel-Permission ^5.9 version
```
into User.php I added `use HasRoles;`
Permissions: `index|create|edit|destroy`
Roles: `user|admin|manager`
All configurations added

**Comment by erikn69** [comment, marked as answer]
Wich line give you that exception?
Maybe [content-must-not-have-property-nor-a-method](https://spatie.be/docs/laravel-permission/v5/prerequisites#content-must-not-have-a-role-or-roles-property-nor-a-roles-method)

**Reply by encoderuz** [reply]
### It works!
Thank you bro! 
In my case there is a `roles` column in the users table )
I remove `roles` column then it works!

**Reply by ghost** [reply]
Thanks for the help!!!
